tools/wake-format/formatter.h

```diff
@@ -117,6 +117,29 @@ inline void newline(wcl::doc_builder& builder, uint8_t space_count) {
   space(builder, space_count);
 }
 
+inline void freshline(wcl::doc_builder& builder, ctx_t ctx) {
+  auto goal_width = SPACE_PER_INDENT * ctx.nest_level;
+  auto merged = ctx.sub(builder);
+
+  // there are non-ws characters on the line, thus a nl is required
+  if (merged->last_width() > merged->last_ws_count()) {
+    newline(builder, goal_width);
+    return;
+  }
+
+  // This is a fresh line, but without the right amount of spaces
+  if (merged->last_width() < goal_width) {
+    space(builder, goal_width - merged->last_width());
+    return;
+  }
+
+  // If there are too many spaces, then a freshline() was used instead
+  // of newline(). Assert to ensure it is fixed.
+  if (merged->last_width() > goal_width) {
+    assert(false);
+  }
```

Comment:
Ah yeah because if its too much and we newline then that's the same as leaving trailing space huh? So we should probs assert in that case...plus since we're not hitting it for right now best to assert.

Reply:
yeah, I don't think it should ever be a valid case to hit this but we can always remove the assert if I am wrong. I want to wait for a valid use case before doing that though